### PR TITLE
Reduce GenISA usages

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Since PoC productization is completed, removing GenISA lowering. 
Some create GenISA static functions are kept in case there are variants we need in the future that are not yet supported by OpenCL C builtins. When that happens, we should report to IGC team asap.

### CODE REVIEW

**Review by etiotto — APPROVED** [review]

**Review by chengjunlu — APPROVED** [review]

**Comment by Dewei-Wang-sh** [comment]
can we keep GenISA? unless opencl is well documented. (which opencl built-in correspond to genISA, what's the naming rule).
right now, I'm still working with genISA.

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp [hunk 1]
```diff
@@ -1013,9 +919,8 @@ struct TritonMatrix2DBlockLoadLowering
       return success();
     }
 
-    // TODO: Remove GenISA lowering after PoC productization is completed.
-    if (tools::getBoolEnv("TRITONGEN_FORCE_GENISA") ||
-        !isOCLBuiltinAvailable(op)) {
+    if (!isOCLBuiltinAvailable(op)) {
+      op.emitWarning("OpenCL API not available for this operation");
```
[the remark this answers is not in the document]
Comment (victor-eds) [reply third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:923]: Not to do in this PR

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp [hunk 2]
```diff
@@ -249,8 +206,9 @@ static bool isOCLBuiltinAvailable(TritonGEN::Matrix2DBlockLoadOp op) {
   return false;
 }
 
-static Value createGenISA2DBlockRead(TritonGEN::Matrix2DBlockLoadOp op,
-                                     ConversionPatternRewriter &rewriter) {
+[[maybe_unused]] static Value
+createGenISA2DBlockRead(TritonGEN::Matrix2DBlockLoadOp op,
+                        ConversionPatternRewriter &rewriter) {
```
[the remark this answers is not in the document]
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:211]: it is, but we already filed IGC ticket for the remaining variants, so adding for the future. Do you refer to add `maybe_unused` in the future when we remove the call to this function?

**Comment by whitneywhtsang** [comment]
> can we keep GenISA? unless opencl is well documented. (which opencl built-in correspond to genISA, what's the naming rule). right now, I'm still working with genISA.

This PR keeps GenISA implementations for 2d block operations, do you need GenISA for other operations?
The cl_intel_subgroup_2d_block_io extension is not yet published, but we have an internal version, could you reference the internal version for documentation in the meantime?

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (victor-eds) [reply third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:211]: Yes, but I'm fine either way

File: third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (victor-eds) [reply third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:923]: > I suppose, but there could be performance impact.

IGC will still split these reads into more than one instruction anyway. Worth checking maybe. As I say, we can go with current state tho

**Review by victor-eds — APPROVED** [review]

**Comment by etiotto** [comment]
Our plan is to reduce (and eventually completely eliminate) usage of GenISA intrinsics in Triton and instead use OpenCL/SPIRV calls. This PR doesn't remove the infrastructure to generate GenISA calls  completely, because we might need to temporarily generate GenISA calls when/if an equivalent OpenCL call is not (yet) available.  @Dewei-Wang-sh I think that is a reasonable approach. Of course we can discuss offline if you have further concerns and/or questions.

**Comment by Dewei-Wang-sh** [comment]
> > can we keep GenISA? unless opencl is well documented. (which opencl built-in correspond to genISA, what's the naming rule). right now, I'm still working with genISA.
> 
> This PR keeps GenISA implementations for 2d block operations, do you need GenISA for other operations? The cl_intel_subgroup_2d_block_io extension is not yet published, but we have an internal version, could you reference the internal version for documentation in the meantime?

yeah, I need genisa for load/store/prefetch2d, dpas, waveall.
the problem is where can I find the documentation?
I'm asking to keep for 2 reasons:
1. we can control all variants in genISA parameter, while opencl need a specific new built-in every time we change a parameter.
this is ugly and we have to wait in certain cases.
2. genIsa has documentation which we can refer to get each parameter's meaning, but opencl do not.

**Comment by Dewei-Wang-sh** [comment]
> Our plan is to reduce (and eventually completely eliminate) usage of GenISA intrinsics in Triton and instead use OpenCL/SPIRV calls. This PR doesn't remove the infrastructure to generate GenISA calls completely, because we might need to temporarily generate GenISA calls when/if an equivalent OpenCL call is not (yet) available. @Dewei-Wang-sh I think that is a reasonable approach. Of course we can discuss offline if you have further concerns and/or questions.

yeah, agreed, keep genISA for temporary usage.
my concern is documentation and transparency, it's hard to use and debug in opencl now.

**Review by Dewei-Wang-sh — APPROVED** [review]
synced offline, I will not update code to use genISA.

**Pull request merged**